network_analysis/find-service-areas-for-multiple-facilities/src/main/java/com/esri/samples/find_service_areas_for_multiple_facilities/FindServiceAreasForMultipleFacilitiesSample.java

```diff
@@ -128,16 +129,17 @@ public void start(Stage stage) {
         mapView.setViewpointGeometryAsync(facilitiesFeatureLayer.getFullExtent(), 130);
 
         // enable the find service areas button when the draw status is completed for the first time
-        mapView.addDrawStatusChangedListener(new DrawStatusChangedListener() {
+        ChangeListener<DrawStatus> drawStatusChangeListener = new ChangeListener<>() {
           @Override
-          public void drawStatusChanged(DrawStatusChangedEvent drawStatusChangedEvent) {
-            if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
+          public void changed(ObservableValue<? extends DrawStatus> observable, DrawStatus oldValue, DrawStatus newValue) {
+            if (newValue == DrawStatus.COMPLETED) {
               // enable the 'find service areas' button
               findServiceAreasButton.setDisable(false);
```

Comment:
Is there some UI we can bind to here - disable property on the button?

Reply:
I tried it but decided not include it as it caused the blinking issue with the button. Currently, the button remains disabled after clicking it once. It is however strange to have all this code for just disabling the button.

Reply:
Binding the button to the feature layer load status works well.